File: MorySnip/Classes/Layers/Layer_Image.vb
Public Class Layer_Image
    Inherits Layer

    Public Sub New(Image As Image)
        Me.Image = Image
        Me.LastPoint = Image.Size
    End Sub

    Public Property Image() As Image

    Public Overrides Sub Paint(g As Graphics)
        g.DrawImage(Me.Image, Me.Bounds)
    End Sub
End Class

File: MorySnip/Classes/Layers/LayerImage.cs
using System.Drawing;

namespace Morysoft.MorySnip
{
    public class LayerImage : Layer
    {
        public LayerImage(Bitmap Image)
        {
            this.Image = Image;
            this.LastPoint = (Point)Image.Size;
        }

        public Bitmap Image { get; set; }

        public override void Paint(Graphics g)
        {
            g.DrawImage(this.Image, this.Bounds);
        }
    }
}
